### PR TITLE
feat: add phase one of periodic evaluation of all MRs in the background

### PR DESCRIPTION
**Pull request by jippi**
**Current limitations**

- It will only work for up to 100 projects, as pagination has not been implemented yet
- It will only ever process the oldest 100 MRs *per project, as pagination has not been implemented yet
- Continuous Evaluation will always use `scm-engine` configuration from `HEAD` (default branch), so MRs with `.scm-engine.yml` changes will not be used

### CODE REVIEW

File: cmd/gitlab_server_periodic.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+				results, err := client.FindMergeRequestsForPeriodicEvaluation(ctx, filter)
+				if err != nil {
+					slogctx.Error(ctx, "Failed to generate merge request list to evaluate", slog.Any("error", err))
+
+					continue
+				}
+
+				for _, mergeRequest := range results {
+					ctx := ctx // make sure we define a
```
Comment (jippi) [inline cmd/gitlab_server_periodic.go:75]: ... fresh context so we don't append to the same context over and over (find link to article about this problem)

**Comment by vschissato** [comment]
LGMT <3

**Review by vschissato — APPROVED** [review]
lgmt

**Review by colinodell — APPROVED** [review]
Nice!

**Pull request merged**